api/v1beta1/gcpmachinetemplate_webhook.go

```diff
@@ -79,11 +77,11 @@ func (r *GCPMachineTemplate) ValidateUpdate(old runtime.Object) error {
 	delete(oldGCPMachineTemplateSpec, "additionalNetworkTags")
 	delete(newGCPMachineTemplateSpec, "additionalNetworkTags")
 
-	if !reflect.DeepEqual(oldGCPMachineTemplateSpec, newGCPMachineTemplateSpec) {
-		return apierrors.NewInvalid(GroupVersion.WithKind("GCPMachineTemplate").GroupKind(), r.Name, field.ErrorList{
-			field.Forbidden(field.NewPath("spec"), "cannot be modified"),
-		})
-	}
+	// if !reflect.DeepEqual(oldGCPMachineTemplateSpec, newGCPMachineTemplateSpec) {
```

Comment:
I think instead of disabling the whole check we should check the fields that are causing this issue and allow changing them by removing them from spec. Please see examples above of how they remove fields that might change